internal/ingress/controller/parser/parser.go

```diff
@@ -1740,26 +1773,55 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+func (p *Parser) kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var err error
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom != (configurationv1.NamespacedSecretValueFromSource{}) {
+		if len(k8sPlugin.Config) > 0 {
+			return kong.Plugin{}, errors.Errorf("cluster plugin '%v' has both Config and ConfigFrom set", k8sPlugin.Name)
+		}
+		config, configError = p.namespacedSecretToConfiguration(k8sPlugin.ConfigFrom)
+		if configError != nil {
+			err = fmt.Errorf("error parsing config for cluster plugin %v: %w", k8sPlugin.Name, configError)
+		}
+		k8sPlugin.Config = config
+	}
+	kongPlugin := toKongPlugin(plugin{
 		Name:   k8sPlugin.PluginName,
-		Config: k8sPlugin.Config,
+		Config: config,
 
 		RunOn:     k8sPlugin.RunOn,
 		Disabled:  k8sPlugin.Disabled,
 		Protocols: k8sPlugin.Protocols,
 	})
+	return kongPlugin, err
 }
 
-func kongPluginFromK8SPlugin(k8sPlugin configurationv1.KongPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+func (p *Parser) kongPluginFromK8SPlugin(k8sPlugin configurationv1.KongPlugin) (kong.Plugin, error) {
+	var err error
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom != (configurationv1.SecretValueFromSource{}) {
+		if len(k8sPlugin.Config) > 0 {
+			return kong.Plugin{}, errors.Errorf("plugin '%v/%v' has both Config and ConfigFrom set",
```

Comment:
`s/plugin/KongPlugin` in log line. Same applies to the code in the clusterplugin